fabric-networking-player-tracking-v1/src/main/java/net/fabricmc/fabric/api/networking/player/tracking/v1/PlayerStream.java

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.networking.player.tracking.v1;
+
+import java.util.Collection;
+import java.util.stream.Stream;
+
+import net.minecraft.block.entity.BlockEntity;
+import net.minecraft.entity.Entity;
+import net.minecraft.server.MinecraftServer;
+import net.minecraft.server.network.ServerPlayerEntity;
+import net.minecraft.server.world.ServerChunkManager;
+import net.minecraft.server.world.ServerWorld;
+import net.minecraft.server.world.ThreadedAnvilChunkStorage;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.ChunkPos;
+import net.minecraft.util.math.Vec3d;
+import net.minecraft.util.math.Vec3i;
+import net.minecraft.world.World;
+import net.minecraft.world.chunk.ChunkManager;
+
+import net.fabricmc.fabric.impl.networking.player.tracking.EntityTrackerStorageAccessor;
+
+/**
+ * Helper streams for looking up players for receiving packets.
+ *
+ * <p>The word "watching" means that an entity/chunk on the server is known to a player's client (in view distance, etc).
+ *
+ * <p>These methods should be called on the server thread and would only work on logical servers.
+ *
+ * <p>For methods returning {@link Collection}s, the returned collections should not be modified in any way.
```

Comment:
Why not use immutable/unmodifiable collections or streams?

Reply:
creating new collections/streams imposes extra overhead. We are not writing an accounting software here.

Also, vanilla already has a great number of dangerous access (e.g. trying to modify the player list from `server.getPlayerManager().getPlayerList()`). Most modders aren't that naughty either.

Reply:
Any more objections on using regular collections? We can just assume most modders are not malicious and breaking the internal state of the game. The safeguard around it would be a bit expensive (performance wise).